src/sentry/incidents/grouptype.py

```diff
@@ -243,15 +249,39 @@ def extract_dedupe_value(self, data_packet: DataPacket[MetricUpdate]) -> int:
     def extract_value(self, data_packet: DataPacket[MetricUpdate]) -> MetricResult:
         # this is a bit of a hack - anomaly detection data packets send extra data we need to pass along
         values = data_packet.packet.values
+
         if isinstance(data_packet.packet, AnomalyDetectionUpdate):
             return {None: values}
-        return values.get("value")
+
+        if "groups" in values:
+            grouped_values = {}
+            for group_data in values["groups"]:
+                group_keys = group_data.get("group_keys", {})
+                group_value = group_data.get("value", 0)
+
+                group_key = self._create_group_key(group_keys)
+                grouped_values[group_key] = group_value
+
+            return grouped_values
+        else:
+            if values.get("value") is not None:
+                return values.get("value")
+            return values
+
+    def _create_group_key(self, group_keys: dict[str, str]) -> str:
+        """Create a deterministic group key from group keys"""
+
+        # Sort keys for deterministic fingerprint.
+        sorted_items = sorted(group_keys.items())
+        key_string = ",".join(f"{k}={v}" for k, v in sorted_items)
+        return key_string
 
     def construct_title(
         self,
         snuba_query: SnubaQuery,
         detector_trigger: DataCondition,
         priority: DetectorPriorityLevel,
+        group_key: DetectorGroupKey | None = None,
```

Comment:
do we also need to update anything related to the fingerprints for groups? i think it'll be important that _each_ detector group key creates a unique fingerprint, so the issues are unique for users.